### PR TITLE
Refactor Overview page.

### PR DESCRIPTION
**Pull request by AngryBerryMS**
Resolve the comments in the merged PR: https://github.com/dotnet/docfx/pull/6571

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6579)

### CODE REVIEW

File: src/docfx/build/toc/TableOfContentsLoader.cs [hunk 1]
```diff
@@ -219,7 +219,7 @@ private void AddOverviewPage(TableOfContentsNode toc)
 
             if (!string.IsNullOrEmpty(toc.Uid) || !string.IsNullOrEmpty(toc.Href))
             {
-                var overview = toc.CloneWithoutItems();
+                var overview = new TableOfContentsNode(toc) { Items = new List<SourceInfo<TableOfContentsNode>>() };
```
Comment (yufeih) [inline src/docfx/build/toc/TableOfContentsLoader.cs:222]:
```suggestion
                var overview = new TableOfContentsNode(toc);
```

File: src/docfx/build/toc/ServicePageGenerator.cs [hunk 2]
```diff
@@ -72,7 +72,8 @@ internal class ServicePageGenerator
                     var childUid = item.Value.Uid.Value;
 
                     if (!string.IsNullOrEmpty(childHref) &&
-                        TableOfContentsLoader.GetHrefType(childHref) == TocHrefType.RelativeFolder)
+                        (TableOfContentsLoader.GetHrefType(childHref) == TocHrefType.RelativeFolder ||
```
Comment (yufeih) [inline src/docfx/build/toc/ServicePageGenerator.cs:75]: consider call `GetHrefType` once

File: src/docfx/build/toc/TableOfContentsLoader.cs (hunk 1 above)
Comment (AngryBerryMS) [reply src/docfx/build/toc/TableOfContentsLoader.cs:222]: For the overview should have only one "Overview" item. Should make the `Items` empty firstly.

File: src/docfx/build/toc/ServicePageGenerator.cs (hunk 2 above)
Comment (AngryBerryMS) [reply src/docfx/build/toc/ServicePageGenerator.cs]: Fixed.

File: src/docfx/build/toc/TableOfContentsLoader.cs (hunk 1 above)
Comment (AngryBerryMS) [reply src/docfx/build/toc/TableOfContentsLoader.cs]: Remove `Items = item.Items` in `TableOfContentsNode.cs` and fix all references of this constructor.

**Review by yufeih — APPROVED** [review]

**Pull request merged**